### ISSUE TITLE
Dependencies set-up, a little messy here

### ISSUE THREAD

**Issue by ychen441** [CognitiveModeling/THICK#1]
Hi Chris @cgumbsch ,

THICK is a masterpiece but I met tons of problems when setting up the dependencies of it. Seems that lots of them got conflicts with others' version, especially numpy, nevergrad, and seaborn. I have modify the requirements by hands, the modified package versions would satisfy installation but not for running the code, e.g., a higher version of numpy does not have "typeDict". Also, some basic problems like "sklearn" and "scikit-learn" appear at the same time here.

Would you mind double-check the .txt when you get time for that? Appreciates.

Below are the .txt I modified without version conflict (BUT not for successfully running any exps!)

**absl-py==0.15.0
ale-py==0.7.5
appdirs==1.4.4
astunparse==1.6.3
atari-py==0.2.9
bayesian-optimization==1.4.3
cachetools==5.3.1
certifi==2023.7.22
charset-normalizer==3.2.0
click==8.1.7
cloudpickle==1.3.0
cma==3.3.0
colorama==0.4.6
contourpy==1.1.1
cycler==0.11.0
decorator==5.1.1
dm-control==1.0.14
dm-env==1.6
dm-tree==0.1.8
docker-pycreds==0.4.0
flatbuffers==1.12
fonttools==4.42.1
future==0.18.3
gast==0.3.3
gitdb==4.0.10
GitPython==3.1.37
glfw==2.6.2
google-auth==2.23.0
google-auth-oauthlib==0.4.6
google-pasta==0.2.0
grpcio==1.32.0
gym==0.17.2
gym-minigrid==1.0.3
h5py==2.10.0
idna==3.4
imageio==2.31.6
importlib-metadata==6.8.0
importlib-resources==6.1.0
joblib==1.3.2
keras==2.8.0rc0
Keras-Preprocessing==1.1.2
kiwisolver==1.4.5
labmaze==1.0.6
lxml==4.9.3
Markdown==3.4.4
MarkupSafe==2.1.3
matplotlib==3.6.0
minihack==0.1.5
mujoco==2.3.7
networkx==3.0
nevergrad==0.13.0
nle==0.9.0
numpy==1.24.1
oauthlib==3.2.2
opencv-python==4.8.0.76
opt-einsum==3.3.0
packaging==23.1
pandas==1.4.4
pathtools==0.1.2
Pillow==9.5.0
promise==2.3
protobuf==3.20.1
psutil==5.9.5
pyasn1==0.5.0
pyasn1-modules==0.3.0
pybind11==2.11.1
pygame==2.5.2
pyglet==1.5.0
PyOpenGL==3.1.7
pyparsing==3.1.1
python-dateutil==2.8.2
pytz==2023.3.post1
pyuv==1.4.0
PyYAML==6.0.1
requests==2.31.0
requests-oauthlib==1.3.1
robodesk==1.0.0
rsa==4.9
ruamel.yaml==0.17.21
ruamel.yaml.clib==0.2.7
scikit-learn==1.3.1
scipy==1.10.1
seaborn==0.12.2
sentry-sdk==1.31.0
setproctitle==1.3.2
shortuuid==1.0.11
six==1.15.0
smmap==5.0.1
tensorboard==2.8
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.1
tensorflow-estimator==2.8.0
tensorflow-gpu==2.8.1
tensorflow-gpu-estimator==2.3.0
tensorflow-io-gcs-filesystem==0.26.0
tensorflow-probability==0.11.0
termcolor==1.1.0
threadpoolctl==3.2.0
tqdm==4.66.1
typing-extensions==3.7.4.3
tzdata==2023.3
urllib3==1.26.16
virtualenv-clone==0.5.7
Werkzeug==2.3.7
wrapt==1.12.1
zipp==3.17.0
**

Best,
Chen

**Comment by ychen441** [comment]
> Hi Chris [@cgumbsch](https://github.com/cgumbsch) ,
> 
> THICK is a masterpiece but I met tons of problems when setting up the dependencies of it. Seems that lots of them got conflicts with others' version, especially numpy, nevergrad, and seaborn. I have modify the requirements by hands, the modified package versions would satisfy installation but not for running the code, e.g., a higher version of numpy does not have "typeDict". Also, some basic problems like "sklearn" and "scikit-learn" appear at the same time here.
> 
> Would you mind double-check the .txt when you get time for that? Appreciates.
> 
> Below are the .txt I modified without version conflict (BUT not for successfully running any exps!)
> 
> **absl-py==0.15.0 ale-py==0.7.5 appdirs==1.4.4 astunparse==1.6.3 atari-py==0.2.9 bayesian-optimization==1.4.3 cachetools==5.3.1 certifi==2023.7.22 charset-normalizer==3.2.0 click==8.1.7 cloudpickle==1.3.0 cma==3.3.0 colorama==0.4.6 contourpy==1.1.1 cycler==0.11.0 decorator==5.1.1 dm-control==1.0.14 dm-env==1.6 dm-tree==0.1.8 docker-pycreds==0.4.0 flatbuffers==1.12 fonttools==4.42.1 future==0.18.3 gast==0.3.3 gitdb==4.0.10 GitPython==3.1.37 glfw==2.6.2 google-auth==2.23.0 google-auth-oauthlib==0.4.6 google-pasta==0.2.0 grpcio==1.32.0 gym==0.17.2 gym-minigrid==1.0.3 h5py==2.10.0 idna==3.4 imageio==2.31.6 importlib-metadata==6.8.0 importlib-resources==6.1.0 joblib==1.3.2 keras==2.8.0rc0 Keras-Preprocessing==1.1.2 kiwisolver==1.4.5 labmaze==1.0.6 lxml==4.9.3 Markdown==3.4.4 MarkupSafe==2.1.3 matplotlib==3.6.0 minihack==0.1.5 mujoco==2.3.7 networkx==3.0 nevergrad==0.13.0 nle==0.9.0 numpy==1.24.1 oauthlib==3.2.2 opencv-python==4.8.0.76 opt-einsum==3.3.0 packaging==23.1 pandas==1.4.4 pathtools==0.1.2 Pillow==9.5.0 promise==2.3 protobuf==3.20.1 psutil==5.9.5 pyasn1==0.5.0 pyasn1-modules==0.3.0 pybind11==2.11.1 pygame==2.5.2 pyglet==1.5.0 PyOpenGL==3.1.7 pyparsing==3.1.1 python-dateutil==2.8.2 pytz==2023.3.post1 pyuv==1.4.0 PyYAML==6.0.1 requests==2.31.0 requests-oauthlib==1.3.1 robodesk==1.0.0 rsa==4.9 ruamel.yaml==0.17.21 ruamel.yaml.clib==0.2.7 scikit-learn==1.3.1 scipy==1.10.1 seaborn==0.12.2 sentry-sdk==1.31.0 setproctitle==1.3.2 shortuuid==1.0.11 six==1.15.0 smmap==5.0.1 tensorboard==2.8 tensorboard-data-server==0.6.1 tensorboard-plugin-wit==1.8.1 tensorflow-estimator==2.8.0 tensorflow-gpu==2.8.1 tensorflow-gpu-estimator==2.3.0 tensorflow-io-gcs-filesystem==0.26.0 tensorflow-probability==0.11.0 termcolor==1.1.0 threadpoolctl==3.2.0 tqdm==4.66.1 typing-extensions==3.7.4.3 tzdata==2023.3 urllib3==1.26.16 virtualenv-clone==0.5.7 Werkzeug==2.3.7 wrapt==1.12.1 zipp==3.17.0 **
> 
> Best, Chen

For anyone who interests, I have got in touch with Chris, and a fix of the dependencies will be addressed in the coming future. I will try to find a compatible one at the same time and publish it here. So I think this might be the end of this issue. Hope you enjoy fiddling with THICK. XD

**Comment by brahmasp** [comment]
Hi @ychen-441 . Thanks for bringing this up! Could you post the updated resolved version? Do you have a Dockerfile for it? Thanks

**Comment by ychen441** [comment]
@brahmasp Hi Brahma, you can find the v3-based THICK here (https://github.com/CognitiveModeling/thix), but the policy right now seems to be a flat one, so you might need to manually add some codes to it. I am working on it, too, hopefully I will make the whole thing right.
